### ISSUE TITLE
TF2.15 Tensorboard-Profiler Errors in Jupyter Notebook

### ISSUE THREAD

**Issue by Te-eMster** [tensorflow/tensorflow#62660]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

TF2.15

### Custom code

No

### OS platform and distribution

Linux Mint 21.2 / Anaconda 23.11

### Python version

3.11

### CUDA/cuDNN version

12.2 / 8.9 ("bundled" versions with tensorflow[and-cuda])

### GPU model and memory

RTX 3080 10GB

### Current behavior?

I installed TF2.15 in a fresh Anaconda-Install as described in the "Install TensorFlow with pip" section. Installation runs smoothly and the example model runs smoothly as well.
The only other installations were:
- conda update conda
- conda install -c "nvidia/label/cuda-12.2.2" cuda-toolkit

However on compiling the model with Tensorboard-Callback in Jupyter Notebook and subsequently fitting the model I get multiple errors. 
First there are the "cuptiGetTimestamp: error 999" and subsequent errors in the logs. Second, when trying to load the Profile-tab in Tensorboard, I get to select Profile, some data is loaded, but it says "Failed to load libcupti (is it installed and accessible?)"

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import os
os.environ['TENSORBOARD_BINARY'] = '/home/user/anaconda3/bin/tensorboard'
os.environ['CUDA_PATH'] = '/home/user/anaconda3'
os.environ['LD_LIBRARY_PATH'] = '/home/user/anaconda3/lib'
os.environ['PATH'] = os.environ['PATH']+':/home/user/anaconda3/lib'
%load_ext tensorboard

mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10)
])
loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
tensorboard_callback = tf.keras.callbacks.TensorBoard(log_dir='/home/user/project/logs/test/test1', histogram_freq = 0, profile_batch='50,100')

model.compile(optimizer='adam',
              loss=loss_fn,
              metrics=['accuracy'])

model.fit(x_train, y_train, epochs=2, callbacks=[tensorboard_callback])

%tensorboard --load_fast=false --logdir='./logs/test/test1'
```

### Relevant log output

```shell
[When creating the "tensorboard_callback" variable]
2023-12-18 16:29:27.188828: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1929] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 7743 MB memory:  -> device: 0, name: NVIDIA GeForce RTX 3080, pci bus id: 0000:02:00.0, compute capability: 8.6
2023-12-18 16:29:27.345796: I external/local_tsl/tsl/profiler/lib/profiler_session.cc:104] Profiler session initializing.
2023-12-18 16:29:27.345815: I external/local_tsl/tsl/profiler/lib/profiler_session.cc:119] Profiler session started.
2023-12-18 16:29:27.345840: I external/local_xla/xla/backends/profiler/gpu/cupti_tracer.cc:1883] Profiler found 1 GPUs
2023-12-18 16:29:27.346062: E external/local_xla/xla/backends/profiler/gpu/cupti_error_manager.cc:137] cuptiGetTimestamp: error 999: 
2023-12-18 16:29:27.346073: E external/local_xla/xla/backends/profiler/gpu/cupti_error_manager.cc:186] cuptiSubscribe: ignored due to a previous error.
2023-12-18 16:29:27.346078: E external/local_xla/xla/backends/profiler/gpu/cupti_error_manager.cc:459] cuptiGetResultString: ignored due to a previous error.
2023-12-18 16:29:27.346082: E external/local_xla/xla/backends/profiler/gpu/cupti_tracer.cc:1935] function cupti_interface_->Subscribe( &subscriber_, (CUpti_CallbackFunc)ApiCallback, this)failed with error 
2023-12-18 16:29:27.346132: I external/local_tsl/tsl/profiler/lib/profiler_session.cc:131] Profiler session tear down.
2023-12-18 16:29:27.346144: E external/local_xla/xla/backends/profiler/gpu/cupti_error_manager.cc:142] cuptiFinalize: ignored due to a previous error.
2023-12-18 16:29:27.346148: E external/local_xla/xla/backends/profiler/gpu/cupti_error_manager.cc:459] cuptiGetResultString: ignored due to a previous error.
2023-12-18 16:29:27.346152: E external/local_xla/xla/backends/profiler/gpu/cupti_tracer.cc:2026] function cupti_interface_->Finalize()failed with error 

[When running model.fit]
2023-12-18 16:29:29.504764: I external/local_xla/xla/service/service.cc:168] XLA service 0x7fc4f9fd6150 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2023-12-18 16:29:29.504788: I external/local_xla/xla/service/service.cc:176]   StreamExecutor device (0): NVIDIA GeForce RTX 3080, Compute Capability 8.6
2023-12-18 16:29:29.510351: I tensorflow/compiler/mlir/tensorflow/utils/dump_mlir_util.cc:269] disabling MLIR crash reproducer, set env var `MLIR_CRASH_REPRODUCER_DIRECTORY` to enable.
2023-12-18 16:29:29.529016: I external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:454] Loaded cuDNN version 8904
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1702913369.646503    9396 device_compiler.h:186] Compiled cluster using XLA!  This line is logged at most once for the lifetime of the process.

[When reaching the profiled batches]
2023-12-18 16:29:30.228094: I external/local_tsl/tsl/profiler/lib/profiler_session.cc:104] Profiler session initializing.
2023-12-18 16:29:30.228137: I external/local_tsl/tsl/profiler/lib/profiler_session.cc:119] Profiler session started.
2023-12-18 16:29:30.228163: E external/local_xla/xla/backends/profiler/gpu/cupti_error_manager.cc:135] cuptiGetTimestamp: ignored due to a previous error.
2023-12-18 16:29:30.228179: E external/local_xla/xla/backends/profiler/gpu/cupti_error_manager.cc:186] cuptiSubscribe: ignored due to a previous error.
2023-12-18 16:29:30.228190: E external/local_xla/xla/backends/profiler/gpu/cupti_error_manager.cc:459] cuptiGetResultString: ignored due to a previous error.
2023-12-18 16:29:30.228201: E external/local_xla/xla/backends/profiler/gpu/cupti_tracer.cc:1935] function cupti_interface_->Subscribe( &subscriber_, (CUpti_CallbackFunc)ApiCallback, this)failed with error
2023-12-18 16:29:30.476342: I external/local_tsl/tsl/profiler/lib/profiler_session.cc:70] Profiler session collecting data.
2023-12-18 16:29:30.478398: E external/local_xla/xla/backends/profiler/gpu/cupti_error_manager.cc:142] cuptiFinalize: ignored due to a previous error.
2023-12-18 16:29:30.478426: E external/local_xla/xla/backends/profiler/gpu/cupti_error_manager.cc:459] cuptiGetResultString: ignored due to a previous error.
2023-12-18 16:29:30.478439: E external/local_xla/xla/backends/profiler/gpu/cupti_tracer.cc:2026] function cupti_interface_->Finalize()failed with error 
2023-12-18 16:29:30.517775: E external/local_xla/xla/backends/profiler/gpu/cupti_error_manager.cc:135] cuptiGetTimestamp: ignored due to a previous error.
2023-12-18 16:29:30.517808: E external/local_xla/xla/backends/profiler/gpu/cupti_error_manager.cc:135] cuptiGetTimestamp: ignored due to a previous error.
2023-12-18 16:29:30.517819: I external/local_xla/xla/backends/profiler/gpu/cupti_collector.cc:541]  GpuTracer has collected 0 callback api events and 0 activity events. 
2023-12-18 16:29:30.531760: I external/local_tsl/tsl/profiler/lib/profiler_session.cc:131] Profiler session tear down.
2023-12-18 16:29:30.533373: I external/local_tsl/tsl/profiler/rpc/client/save_profile.cc:144] Collecting XSpace to repository: /home/user/project/logs/test/test1/plugins/profile/2023_12_18_16_29_30/user.xplane.pb
```

**Comment by sushreebarsa** [comment]
@Te-eMster Could you please verify CUPTI Installation and Compatibility using the following:
`conda list cupti` also please ensure `LD_LIBRARY_PATH `includes the path to CUPTI libraries(e.g.,/usr/local/cuda/extras/CUPTI/lib64).
Thank you!

**Comment by Te-eMster** [comment]
@sushreebarsa Thanks for your help.

Here are the relevant outputs from terminal:

**conda list cupti in kernel-environment:**
# packages in environment at /home/matai/anaconda3/envs/tf2.15-py3.11:          
#                                                                               
# Name                    		Version                   Build  	Channel              
cuda-cupti                		12.2.142                      0    	nvidia/label/cuda-12.2.2                                                                           
cuda-cupti-static         		12.2.142                      0    	nvidia/label/cuda-12.2.2                                                                           
nvidia-cuda-cupti-cu12    	12.2.142                 pypi_0    	pypi   

**"os.environ.__getitem__('LD_LIBRARY_PATH‘)" executed in the Notebook:**
'/home/user/anaconda3/envs/tf2.15-py3.11/lib'

**"ls | grep libcupti" executed in the LD_LIBRARY_PATH directory:**
libcupti.so
libcupti.so.12
libcupti.so.2023.2.2
libcupti_static.a

**Comment by sushreebarsa** [comment]
@Te-eMster Could you please have a look at this colab [gist](https://colab.research.google.com/gist/sushreebarsa/5ad09985df807cf6938748ab83f9d917/62660.ipynb) where I was able to run the provided code successfully ?
Please create a new environment and try again. Thank you!

**Comment by giulatona** [comment]
I was having the same issue. I had installed tensorlow[and-cuda] using pipenv on a Ubuntu 22.04 machine.
I solved it by adding "_env_path_/lib/python3.10/site-packages/nvidia/cuda_cupti/lib/" to LD_LIBRARY_PATH. 

Is there a way to avoid manually setting LD_LIBRARY_PATH ?
